rust/src/traits/samplers/discretize/mod.rs

```diff
@@ -215,17 +230,36 @@ mod test {
     }
 
     #[test]
-    fn test_rational_k() -> Fallible<()> {
-        println!("{:?}", 1.5.into_rational()? << -2);
+    fn test_extreme_rational() -> Fallible<()> {
+        // rationals with greater magnitude than MAX saturate to infinity
+        let rat = RBig::try_from(f64::MAX).unwrap();
+        assert!((rat * IBig::from(2u8)).to_f64().value().is_infinite());
+
         Ok(())
     }
 
     #[test]
-    fn test_extreme_rational() -> Fallible<()> {
-        // rationals with greater magnitude than MAX saturate to infinity
-        let rat = Rational::from_f64(f64::MAX).unwrap();
-        println!("{:?}", (rat * 2u8).to_f64());
+    fn test_shr() -> Fallible<()> {
+        assert_eq!(shr(RBig::try_from(1.)?, 0), RBig::from(1u8));
+        assert_eq!(shr(RBig::try_from(0.25)?, -2), RBig::from(1u8));
+        assert_eq!(shr(RBig::try_from(1.)?, 2), RBig::try_from(0.25)?);
+        Ok(())
+    }
 
+    #[test]
+    fn test_find_nearest_multiple_of_2k() -> Fallible<()> {
+        assert_eq!(
+            find_nearest_multiple_of_2k(RBig::try_from(-2.25)?, 0),
+            IBig::from(-2)
+        );
+        assert_eq!(
+            find_nearest_multiple_of_2k(RBig::try_from(2.25)?, -1),
+            IBig::from(5)
```

Comment:
I don't understand this. Docstring:
```
/// Find index of nearest multiple of $2^k$ from x.
```
So I'd interpret this as finding the nearest multiple of `2^-1 = 1/2` to 2.25, ie, either 2 or 2.5. In either case, not 5.